lib/linked_in/images.rb

```diff
@@ -0,0 +1,42 @@
+require 'open-uri'
+
+module LinkedIn
+  #  Images API
+  # @see https://learn.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/images-api
+  class Images < APIResource
+    #  Initializing the upload
+    #  @see https://learn.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/images-api?view=li-lms-2023-05&tabs=http#initialize-image-upload
+    #  @options options [String] :owner, the urn of the owner of the image
+    def upload_url(options = {})
+      path = "rest/images?action=initializeUpload"
+      response = post(path, params(options))
+      body = JSON.parse(response.body)
+      value = body["value"]
+      { url: value['uploadUrl'], urn: value['image'] }
+    end
+
+    #  Uploading the Image
+    #  Note: `Content-Type` header although not mentioned in the docs is required, the upload fails with 400 without it.
+    #  @see https://learn.microsoft.com/en-us/linkedin/marketing/integrations/community-management/shares/vector-asset-api?view=li-lms-2023-05&tabs=http#upload-the-image
+    #  @options options [String] :owner, the urn of the owner of the image
+    def upload_image(image, options)
+      response = upload_url(options)
+      body = file(image)
+      @connection.put(response[:url], body) do |req|
+        req.headers['Content-Type'] = 'image/png'
```

Comment:
Hey, @imanpalsingh Should we use `Content-Type` as `multipart/form-data`? `image/png` feels like we'd always send a png image

Reply:
`multipart/form-data` caused the uploads to fail. Using `image/*` as mime type for uploads. https://github.com/marlytics/linkedin-v2/pull/4